ext/nnstreamer/tensor_filter/tensor_filter_tensorflow_lite.cc

```diff
@@ -495,9 +484,53 @@ TFLiteInterpreter::moveInternals (TFLiteInterpreter& interp)
 #ifdef ENABLE_TFLITE_NNAPI_DELEGATE
   nnfw_delegate = std::move (interp.nnfw_delegate);
 #endif
+  inputTensorPtr = std::move (interp.inputTensorPtr);
+  outputTensorPtr = std::move (interp.outputTensorPtr);
   setModelPath (interp.getModelPath ());
 }
 
+/**
+ * @brief cache input and output tensor ptr before invoke
+ * @return 0 on success. -errno on failure.
+ */
+int
+TFLiteInterpreter::cacheInOutTensorPtr ()
+{
+  int tensor_idx;
+  TfLiteTensor *tensor_ptr;
+
+  inputTensorPtr.clear ();
+  inputTensorPtr.reserve (inputTensorMeta.num_tensors);
+  for (unsigned int i = 0; i < inputTensorMeta.num_tensors; ++i) {
+    tensor_idx = interpreter->inputs ()[i];
```

Comment:
How did you verify that we may cache interpreter->inputs() and outputs()? Is this usage guaranteed by tf-lite?

Reply:
I am not sure about mutliple instances of invoke() as tflite documentation says and I qoute **[An interpreter must not be accessed from concurrent threads.](https://www.tensorflow.org/lite/guide/inference)**.
Regarding the reuse, interpreter builds the graph with nodes wher allocation of nodes and its memories takes place with `interpreter->AllocateNodes()`.
`interpreter->inputs()` and `interpreter->outputs()` provides graph nodes (where the graph remains fixed once delegates is set, tensors are allocated) which also remain fixed.

However, apart from all said above, they do not explicitly mention that `inputs()` and `outputs()` can be cached or its usage. But they do say [Memory allocation for input and output tensors must be triggered by calling AllocateTensors() right after resizing tensors.](https://www.tensorflow.org/lite/guide/inference), which does show that `AllocateTensors()` sets and allocates the input and output tensors.

Reply:
I'm not concerned with simultaneous usage. I'm concerned with the sequential usage of cached tensor pointers.
However, with Line 201 and 206, it appears safe.